framework/infrastructure/traffic_director.py

```diff
@@ -679,32 +724,60 @@ def find_unused_forwarding_rule_port(
         # TODO(sergiitk): custom exception
         raise RuntimeError("Couldn't find unused forwarding rule port")
 
-    def create_forwarding_rule(self, src_port: int):
-        name = self.make_resource_name(self.FORWARDING_RULE_NAME)
+    def create_forwarding_rule(
+        self, src_port: int, ipv6=False, ip_address="0.0.0.0"
```

Comment:
We should probably add type hints here since we have it for `src_port`.

Reply:
Not present anymore after refactor